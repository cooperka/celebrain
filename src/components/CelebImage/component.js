// @flow

import React from 'react';
import { withStyles } from 'material-ui/styles';

import './styles.css';

type Props = {
  classes: any,
  celeb: any,
  noLink: boolean,
};

const styles = (theme) => ({
  shadowed: {
    boxShadow: theme.shadows[2],
  },
});

function CelebImage(props: Props) {
  const { classes, celeb, noLink } = props;

  const image = (
    <img
      className={classes.shadowed}
      src={celeb.image}
      alt="Celebrity"
    />
  );

  return noLink ? image : (
    <a href={`https://commons.wikimedia.org/wiki/File:${celeb.filename}`}>
      {image}
    </a>
  );
}

export default withStyles(styles)(CelebImage);
