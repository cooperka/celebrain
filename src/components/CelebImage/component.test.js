// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import imageData from '../../celebs.json';

import CelebImage from './component';

it('renders', () => {
  const tree = renderer.create(<CelebImage celeb={imageData[0]} />);
  expect(tree.toJSON()).toMatchSnapshot();
});
