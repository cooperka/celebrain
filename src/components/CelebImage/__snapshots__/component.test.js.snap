// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<a
  href="https://commons.wikimedia.org/wiki/File:Arnold_Schwarzenegger_in_Film_Festival.jpg"
>
  <img
    alt="Celebrity"
    className="CelebImage-shadowed-1"
    src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/1c/Arnold_Schwarzenegger_in_Film_Festival.jpg/800px-Arnold_Schwarzenegger_in_Film_Festival.jpg"
  />
</a>
`;
